src/util/formats/adapters/rmd-adapter.ts

```diff
@@ -0,0 +1,117 @@
+import type { FileAdapter } from '../adapter-format';
+import fs from 'fs';
+import { Parser } from 'commonmark';
+import matter from 'gray-matter';
+import { guard } from '../../assert';
+
+export interface CodeBlock {
+	options: string,
+	code:    string,
+}
+
+type CodeBlockEx = CodeBlock & {
+	startpos: { line: number, col: number }
+}
+
+export interface RmdInfo {
+	blocks:  CodeBlock[]
+	options: object
+}
+
+export const RmdAdapter = {
+	readFile: (p: string) => {
+		// Read and Parse Markdown
+		const raw = fs.readFileSync(p, 'utf-8').toString();
+		const parser = new Parser();
+		const ast = parser.parse(raw);
+		
+		// Parse Frontmatter
+		const frontmatter = matter(raw);
+
+		// Parse Codeblocks
+		const walker = ast.walker();
+		const blocks: CodeBlockEx[] = [];
+		let e;
+		while((e = walker.next())) {
+			const node = e.node;
+			if(node.type === 'code_block') {
+				if(!node.info || 
+				   !node.literal ||
+				   !node.info.startsWith('{r') || 
```

Comment:
isn't capital R allowed as well? 
additionally: this will trigger for: `{rust` won't it?